### PR TITLE
Fix compilation on OCaml 4.14.

### PR DESCRIPTION
**Pull request by rlepigre**
This addresses https://github.com/LPCIC/elpi/issues/139.

### CODE REVIEW

File: elpi.opam [hunk 1]
```diff
@@ -14,8 +14,9 @@ build: [
 ]
 
 depends: [
-  "ocaml" {>= "4.04.0" & < "4.14.0"}
+  "ocaml" {>= "4.04.0" & < "4.15.0"}
```
Comment (gares) [inline elpi.opam:17]:
```suggestion
  "ocaml" {>= "4.04.0"}
```

**Comment by gares** [comment]
thanks

**Comment by gares** [comment]
Something very fishy ;-)
```
Error: This expression has type char Stream.t
but an expression was expected of type char Stream.t
```

**Comment by rlepigre** [comment]
> Something very fishy ;-)
> 
> ```
> Error: This expression has type char Stream.t
> but an expression was expected of type char Stream.t
> ```

Oh my... That's annoying. I guess that's what you get when there are two `Stream` modules around. :)
No idea how to fix that though, `camlp5` is probably using the module from the standard library.

I guess the point of https://github.com/LPCIC/elpi/pull/130 is to avoid doing this altogether, right? In that case, I guess we can just wait for https://github.com/LPCIC/elpi/pull/130, since it seems you already made good progress there @gares. ;)

**Comment by rlepigre** [comment]
(Perhaps we could try to tweak the linking order, but that would probably be fragile.)

**Pull request closed without merging**